### PR TITLE
Updating ember responsive and some syntax

### PR DESCRIPTION
**Pull request by jrock2004**
Fixes empress/empress-blog/issues/126

Ember responsive image has changed the way to use it. Followed https://github.com/kaliber5/ember-responsive-image/blob/master/docs/MIGRATION.md for how to migrate. Also in templates I touched, I updated some syntax to octane syntax. 

Big change here is config for ember responsive should be in `ember-cli-build` instead of `config/environment`.

### CODE REVIEW

File: index.js
Comment (elwayman02) [inline index.js:29]: @mansona can you comment on whether we still need this?

**Review by elwayman02 — APPROVED** [review]
The deploy preview for this looks great, thanks for taking this on! I'll defer to Chris' opinion on removing that config from index.js, but otherwise I think this is ready to go.

File: blueprints/empress-blog-casper-template/index.js
Comment (elwayman02) [inline blueprints/empress-blog-casper-template/index.js:18]: Nice work figuring this part out!

File: index.js
Comment (mansona) [reply index.js:29]: So we will need this and we should probably remove the code from the ember-cli-build file.

Ember-cli-build only works on the dummy app when working in an add-on, if we relied on only this they we would break a lot of downstream setups 🙈 does this make sense?

Comment (elwayman02) [reply index.js:29]: Hmm, but ember-responsive-image changed where the config lives in v2, so wouldn't that impact us @mansona?

> The configuration of the addon moved from config/environment.js to ember-cli-build.js.

https://github.com/kaliber5/ember-responsive-image/blob/v2.0.0/docs/MIGRATION.md

**Comment by elwayman02** [comment]
@jrock2004 we're going to take some time to chat about the config issue, since there are some complexities involved. I don't want you to think we're ignoring the PR, but it might sit for a bit while we figure out how to proceed. If you'd like to split out the syntax changes that were unrelated to the ember-responsive-image upgrade, I'd be happy to review and merge those separately so they don't have to wait on the config issue.

File: index.js
Comment (jrock2004) [reply index.js:29]: If this is the code I actually committed in ember-cli-build itself, I committed those so for future developers who work on it can run the dummy app and not get errors

**Comment by jrock2004** [comment]
> @jrock2004 we're going to take some time to chat about the config issue, since there are some complexities involved. I don't want you to think we're ignoring the PR, but it might sit for a bit while we figure out how to proceed. If you'd like to split out the Octane syntax changes that were unrelated to the ember-responsive-image upgrade, I'd be happy to review and merge those as a separate PR so they don't have to wait on the config issue.

Sure, I can go back and undo octane stuff. I just did it cause the errors in my editor were calling me name to fix it :)

**Comment by elwayman02** [comment]
> > @jrock2004 we're going to take some time to chat about the config issue, since there are some complexities involved. I don't want you to think we're ignoring the PR, but it might sit for a bit while we figure out how to proceed. If you'd like to split out the Octane syntax changes that were unrelated to the ember-responsive-image upgrade, I'd be happy to review and merge those as a separate PR so they don't have to wait on the config issue.
> 
> Sure, I can go back and undo octane stuff. I just did it cause the errors in my editor were calling me name to fix it :)

They're much needed improvements in this repo for sure! I just don't want all that work you did to get stalled while we solve the rest of the PR. :) Really appreciate your hard work here.

**Comment by jrock2004** [comment]
Octane syntax has been removed

File: index.js
Comment (elwayman02) [reply index.js:29]: Basically we want to solve this issue before we merge this: https://github.com/empress/empress-blog-casper-template/pull/37

We want ember-responsive-image to be a dependency of the casper template without having to generate the config in the consuming app.

File: blueprints/empress-blog-casper-template/index.js
Comment (jrock2004) [reply blueprints/empress-blog-casper-template/index.js:18]: I think in the future we should figure some of the numbers here and see if we would want to change add or remove some of them

File: index.js
Comment (jrock2004) [reply index.js:29]: Would you want to do that though? It seems like this project has to write the CSS to fit the images. If you pass this on to another app and they specify a config that is too big or small, it could break the templates. Or do you think this would not be an issue

Comment (elwayman02) [reply index.js:29]: No, that's the point. Right now, today, apps can specify any config they want. The config lives in the app and is provided by the blueprint for the template. We want to bring that internally to this project so that the casper template controls the config and can have the correct settings for the layouts.

Comment (jrock2004) [reply index.js:29]: Is that because it was in the config/env? Maybe now that is in cli build they cannot? 🤔

Comment (elwayman02) [reply index.js:29]: This ember-cli-build file actually only runs for the demo app, it doesn't impact consumers. That's why, to maintain the existing functionality, you had to switch to that `applyBuildConfig` method in the blueprint. We basically want to get rid of that altogether and find a way for consumers to inherit a working config without writing anything to the consuming projects where this is installed.

**Comment by NullVoxPopuli** [comment]
> Octane syntax has been removed

Seems like lost effort, imo. 

I know folks are strict about tiny PRs, but if a PR is good, and at the very least, work is split up via commits, why would it be hard to review? 🙃

**Comment by elwayman02** [comment]
> > Octane syntax has been removed
> 
> Seems like lost effort, imo.
> 
> I know folks are strict about tiny PRs, but if a PR is good, and at the very least, work is split up via commits, why would it be hard to review? 🙃

@NullVoxPopuli this had nothing to do with it being hard to review and everything to do with the fact that this PR might take a while to land (due to issues related specifically to the ember-responsive-images config). If the Octane changes were submitted as a separate PR we could merge that first rather than waste the effort put into converting it by leaving it stalled in this PR, which likely isn't going to be merged quickly.

**Comment by elwayman02** [comment]
Closing because this was superseded by #37 & #45. Thanks again for being the driver to get this fixed! I'm still happy to review and merge any PRs for the Octane syntax changes, whenever you have time. :)

**Pull request closed without merging**
